src/modules/commander/state_machine_helper.cpp

```diff
@@ -923,8 +931,7 @@ bool set_nav_state(struct vehicle_status_s *status,
 		if (status->engine_failure) {
 			status->nav_state = vehicle_status_s::NAVIGATION_STATE_AUTO_LANDENGFAIL;
 
-		} else if (status_flags->gps_failure || (!status_flags->condition_global_position_valid ||
-				!status_flags->condition_home_position_valid)) {
+		} else if (status_flags->gps_failure || !status_flags->condition_local_position_valid) {
```

Comment:
In a GPS-denied scenario, this condition will fail. We might have no GPS, but local position might still be fine.